fuel-core/src/cli/run/p2p.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    net::{IpAddr, Ipv4Addr},
+    path::PathBuf,
+    str::FromStr,
+    time::Duration,
+};
+
+use clap::Args;
+
+use fuel_p2p::{Keypair, Multiaddr, P2PConfig, PeerId};
+
+#[derive(Debug, Clone, Args)]
+pub struct P2pArgs {
+    /// Path to the location of DER-encoded Secp256k1 Keypair
+    #[clap(long = "keypair")]
+    pub keypair: Option<PathBuf>,
+
+    /// The name of the p2p Network
+    #[clap(long = "network")]
+    pub network: String,
+
+    /// p2p network's IP Address
+    #[clap(long = "address")]
+    pub address: Option<IpAddr>,
+
+    /// p2p network's TCP Port
+    #[clap(long = "port", default_value = "4000")]
+    pub port: u16,
```

Comment:
This conflicts with the graphql port cli arg. Attempting to start fuel-core causes the following error:
```
thread 'main' panicked at 'Command fuel-core: Argument names must be unique, but 'port' is in use by more than one argument or group'
```

```suggestion
    #[clap(long = "peering-port", default_value = "4000")]
    pub port: u16,
```